src/anchoredinterval.jl

```diff
@@ -122,16 +133,31 @@ end
 # can get unexpected behaviour if adding the span to the anchor endpoint produces a value
 # that is no longer comparable (e.g., `NaN`).
 
-function Base.first(interval::AnchoredInterval{P}) where P
-    P < zero(P) ? (interval.anchor + P) : (interval.anchor)
+function Base.first(interval::AnchoredInterval{P,T,L,R}) where {P,T,L,R}
+    return if L !== Unbounded
+        P < zero(P) ? (interval.anchor + P) : (interval.anchor)
+    else
+        nothing
+    end
 end
 
-function Base.last(interval::AnchoredInterval{P}) where P
-    P < zero(P) ? (interval.anchor) : (interval.anchor + P)
+function Base.last(interval::AnchoredInterval{P,T,L,R}) where {P,T,L,R}
+    return if R !== Unbounded
+        P < zero(P) ? (interval.anchor) : (interval.anchor + P)
+    else
+        nothing
+    end
 end
 
 anchor(interval::AnchoredInterval) = interval.anchor
-span(interval::AnchoredInterval{P}) where P = abs(P)
+
+function span(interval::AnchoredInterval{P}) where P
+    if !isunbounded(interval)
+        abs(P)
+    else
+        throw(ArgumentError("Unable to determine the span of an unbounded interval"))
```

Comment:
Maybe should be a `DomainError`?

Reply:
Could be either I think

Reply:
I've updated to use a `DomainError`

Reply:
See https://github.com/invenia/Intervals.jl/pull/104/commits/59ce3fe32cd9227d0410cafdce72a9aa6d038a57